### DISCUSSION TITLE
Ability to Copy + paste / duplicate sketches

### DISCUSSION THREAD

**Discussion by dmatscheko** [hlorus/CAD_Sketcher#266] [Feature Requests]
**Motivation / Existing Problem**

I want to be able to copy + paste / duplicate sketches. This could also copy the generated object's modifiers.

Been working with CAD Sketcher for three weeks now and the only thing I really miss (or don't know how to do) is duplicating or copying + pasting a sketch. This is important if you need several slightly different variants of a sketch, or if you want to split a sketch into several.

**Description**

- In the List of sketches, select a sketch via mouse or keyboard. Then click `<Shift>+<D>` to duplicate the selected sketch.
and
- In the List of sketches, select a sketch via mouse or keyboard. Then click `<Ctrl>+<C>`, and then `<Ctrl>+<V>` to duplicate the selected sketch.
and
- In the List of sketches, right click a sketch via the mouse. In the context menu, there should be an option to duplicate the sketch.

Each of the above options should generate a duplicate of the selected sketch. Optionally, the modifiers of the object created by the selected sketch can also be duplicated.

**Alternative Solutions**

1. The behavior of feature request https://github.com/hlorus/CAD_Sketcher/issues/14 is not the same, because it wants to basically make a linked object, which is already possible. If the feature request is implemented as creating a copy of the sketch and not a link, it should be possible to break the connection between a sketch and its object, to allow further modification of the object via default-Blender features.
2. Another good feature would be if you could copy and paste points and lines within and between sketches. This could also be used to duplicate the entire content of one sketch into another. To do this, the key combination `<A>` to select everything could also be implemented in a sketch.

**Comment by dtremain** [comment]
My examples are: I want bores with different sized cores & X/Y dimensions. Rather than have to recreated the entire sketch from scratch each time, I would love to be able to duplicate a sketch, and simply edit the parameters to get the new size. Another example would be using cross bore / shaft connectors - the size of the base for the connector may be different, but the bore/shaft needs to be the same size. Again, pain in the tail to re-create the entire sketch for each variation vs. copying the sketch, and making some minor changes.

**Comment by Shaddy-ElDesouky** [comment]
any news on duplicating sketches or copy paste, I see it is on the roadmap for 2.0. I'm currently running 0.27.5 on blender 4.2

**Reply by hlorus** [reply]
Copy&Paste is supported for quite a while already. To duplicate a sketch you'll have to select everything and manually copy it over to a new sketch.

**Comment by Shaddy-ElDesouky** [comment]
@hlorus I tried, it works fine, good for now, thanks a lot for the fast response.
